CMakeLists.txt

```
@@ -134,9 +134,10 @@ include(CheckAssembler)
 include(CheckIntrinsics)
 
 #---Enable asserts------------------------------------------------------------------------------
-if(NOT asserts)
-  set(CMAKE_CXX_FLAGS_${BUILD_TYPE} "${CMAKE_CXX_FLAGS_${BUILD_TYPE}} -DNDEBUG")
-  set(CMAKE_C_FLAGS_${BUILD_TYPE} "${CMAKE_C_FLAGS_${BUILD_TYPE}} -DNDEBUG")
+if(asserts)
+  string(TOUPPER BUILD_TYPE ${CMAKE_BUILD_TYPE})
+  string(REGEX REPLACE "-.NDEBUG" "" CMAKE_CXX_FLAGS_${BUILD_TYPE} "${CMAKE_CXX_FLAGS_${BUILD_TYPE}}")
+  string(REGEX REPLACE "-.NDEBUG" "" CMAKE_C_FLAGS_${BUILD_TYPE} "${CMAKE_C_FLAGS_${BUILD_TYPE}}")
```

Why can I not `-INDEBUG`? Why do you break my favorite CXX flag `-DNDEBUG_EVERYTHING=42`? ;-) What about `-DNDEBUG=1`?

```suggestion
  string(REGEX REPLACE "-[DU]NDEBUG " "" CMAKE_CXX_FLAGS_${BUILD_TYPE} "${CMAKE_CXX_FLAGS_${BUILD_TYPE}}")
  string(REGEX REPLACE "-[DU]NDEBUG " "" CMAKE_C_FLAGS_${BUILD_TYPE} "${CMAKE_C_FLAGS_${BUILD_TYPE}}")
```

Maybe better (since CMake 3.6):
```suggestion
  list(FILTER CMAKE_CXX_FLAGS_${BUILD_TYPE} EXCLUDE REGEX "-[UD]NDEBUG(=.*)?") 
  list(FILTER CMAKE_C_FLAGS_${BUILD_TYPE} EXCLUDE REGEX "-[UD]NDEBUG(=.*)?") 
```

---

Ah, true. I didn't know that people could have those as their favourite flags.

My preferred solution is to use
`cmake -DCMAKE_CXX_FLAGS_RELEASE=-O3 <src>` - less CMake code on our side.

If `-Dasserts=On` is desired, though, we can extend that regex.

---

These are not lists, so have to go for the other solution.